pandas/core/categorical.py

```diff
@@ -1388,15 +1388,22 @@ def unique(self):
 
         Unused categories are NOT returned.
 
+        For ordered categoricals, the unique values are returned in order.
+        For unordered categoricals, the unique values are returned in order of
+        appearance.
+
         Returns
         -------
         unique values : array
         """
-        unique_codes = np.unique(self.codes)
-        # for compatibility with normal unique, which has nan last
-        if unique_codes[0] == -1:
-            unique_codes[0:-1] = unique_codes[1:]
-            unique_codes[-1] = -1
+        from pandas.core.nanops import unique1d
+        unique_codes = unique1d(self.codes)
```

Comment:
A comment on the difference between `np.unique` and `unique1d` and why the latter is used would be nice.